src/targets/services/helpers/model.js

```diff
@@ -14,15 +17,94 @@ export class Model {
     this.classifiers = []
 
     // Using map to allocate a new array for each line
-    this.occurences = Array(vocabulary.length)
+    this.occurences = Array(this.uniqueY.length)
       .fill(0)
-      .map(() => Array(this.uniqueY.length).fill(0))
-    this.logProbabilities = Array(vocabulary.length)
+      .map(() => Array(vocabulary.length).fill(0))
+    this.logProbabilities = Array(this.uniqueY.length)
       .fill(0)
-      .map(() => Array(this.uniqueY.length).fill(0))
+      .map(() => Array(vocabulary.length).fill(0))
     this.contributions = 1
   }
 
+  /**
+   * Internal function used to initialize the model given occurences
+   *
+   * @private
+   */
+   initializeLogProbabilities() {
```

Comment:
Why log-proba and not actual proba? I think cozy-konnector-libs computes actual proba?